scripts/junit2issue.py

```diff
@@ -0,0 +1,189 @@
+        print("skipped = {}".format(suite.skipped))
+
+        for prop in suite.child(Properties):
+            print("{} : {}".format(prop.name, prop.value))
+            version = prop.value
+
+        for case in suite:
+            # handle cases
+            my_case = ZephyrTestCase.fromelem(case)
+            for res in my_case.result:
+                if isinstance(res, test_result):
+                    print("+")
+                    print("  " + my_case.classname)
+                    print("  " + my_case.name)
+                    print("  " + res.message)
+                    report_case = {
+                        'APP_NAME' : my_case.classname,
+                        'CASE_NAME' : my_case.name,
+                        'ZEPHYR_VERSION' : version,
+                        'RESULT' : res.message,
+                        'PLATFORM': suite.name,
+                        'LOGS': res.text,
+                    }
+                    report_list.append(report_case)
+                    # print("  " + res.text)
+    return report_list
+
+def gen_content_from_template(template, context):
+    '''
+        gen_content_from_template
+        @template: in template string
+        @context: in context
+        return:
+            generated issue content string
+    '''
+    _s = Template(template)
+    content = _s.substitute(context)
+    # logging.info(content)
+    return content
+
+def github_login():
+    '''
+        github_login
+    '''
+
+    access_token = os.getenv('GITHUB_ACCESS_TOKEN')
+
+    if not access_token:
+        print("no access token find in system environment variable\n")
+        print("please set GITHUB_ACCESS_TOKEN with:\n")
+        print("export GITHUB_ACCESS_TOKEN=<your github access token>\n")
+        return None
+
+    # using an access token
+    return Github(access_token)
+
+def check_issue_existing(repository, report):
+    '''
+        check whether the report issue exists
+        @repository: in repo target
+        @report: dict fo reporting content
+        return:
+            existing issue or None
+    '''
+    open_issues = repository.get_issues(state='open')
+    for _issue in open_issues:
+        if report['CASE_NAME'] in _issue.title:
+            return _issue
+        if report['LOGS'] in _issue.body:
+            return _issue
+
+    return None
+
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.DEBUG,
+                        format='[%(filename)s:%(lineno)d] %(threadName)s %(message)s')
+    pfile = 'lpcxpresso55s28.xml'
+    vfolder = 'zephyr-v2.6.0-286-g46029914a7ac'
+    auto_report = False
+    try:
+
+        opts, args = getopt.getopt(sys.argv, "hav:i:", ["vfolder=", "pfile="])
+        for opt, arg in opts:
+            if opt == '-h':
+                print('junit2issue.py -v <vfolder> -i <pfile>')
+                sys.exit(0)
+            elif opt in ("-i", "--pfile"):
+                pfile = arg
+            elif opt in ("-v", "--vfolder"):
```

Comment:
I don't think we need “vfolder” option. The only "--pfile" is enough. We tell the user that a full path should be added with "pfile" option.

Reply:
it is reserved for loop a folder. although I did not implement them yet.